docs/model.md

```diff
@@ -0,0 +1,122 @@
+
+
+NOTICE: this document represents the state of the model at the next Heapster release, and is not representative of the current state of the master branch
+TODO(afein): remove NOTICE prior to next release
+
+## Usage
+
+The Heapster Model can be enabled by initializing Heapster with the `-use_model=true` flag. The resolution of the model can be configured through
+the `-model_resolution` flag, which will cause the model to store historical data at the specified resolution. If the `model_resolution` flag is not specified, the default
+resolution of 1 minute will be used.
+
+## API documentation
+
+A detailed documentation of each API endpoint is listed below. 
+
+All endpoints ending in `/metrics/{metric-name}` can accept the optional `start` and `end` query parameters 
+that represent the start and end time of the requested timeseries. The result
+will be a list of (Timestamp, Value) pairs in the time range (start, end].
+`start` and `end` are strings formatted according to RFC3339. If `start` is not
+defined, it is assumed as the zero Unix epoch time. If `end` is not defined,
+then all data later than `start` will be returned.
+
+### Cluster-level Metrics and Stats
+`/api/v1/model/`: The root path of the model API, displays all browsable subpaths.
+
+`/api/v1/model/metrics/`: Displays a list of available cluster-level metrics.
+
+`/api/v1/model/metrics/{metric-name}?start=X&end=Y`: Displays a set of (Timestamp, Value) 
+pairs for the requested cluster-level metric, between the time range specified by `start` and `end`. 
+
+`/api/v1/model/stats/`: Exposes the average, max and 95th percentile over the
+past minute, hour and day for each cluster metric.
+
+### Node-level Metrics and Stats
+`/api/v1/model/nodes`: Displays a list of all available nodes, along
+with their latest CPU and Memory Usage values.
+
+`/api/v1/model/nodes/{node-name}`: Displays all browsable subpaths for a
+specific node.
+
+`/api/v1/model/nodes/{node-name}/pods`: Displays a list of all available pods
+under a given node, along with their latest CPU and Memory Usage values
+
+`/api/v1/model/nodes/{node-name}/metrics/`: Displays a list of available
+node-level metrics
+
+`/api/v1/model/metrics/nodes/{node-name}/{metric-name}?start=X&end=Y`: Displays a set of (Timestamp, Value) 
+pairs for the requested node-level metric, within the time range specified by `start` and `end`. 
+
+`/api/v1/model/nodes/{node-name}/stats/`: Exposes the average, max and 95th
+percentile over the past minute, hour and day for each node metric
+
+
+### Namespace-level Metrics and Stats
+`/api/v1/model/namespaces`: Displays a list of all available namespaces, along
+with their latest CPU and Memory Usage values.
+
+`/api/v1/model/namespaces/{namespace-name}`: Displays all browsable subpaths for
+a specific namespace.
+
+`/api/v1/model/namespaces/{namespace-name}/metrics/`: Displays a list of available namespace-level metrics
```

Comment:
That is a more complicated use case, assuming you mean that they want exactly the same `start` and `end` times for each metric. 
As for now they can call each individual metric's API with the desired start and end.

To enrich the API, we could add a model method that extracts all metrics for an entity given the same `start` and `end` times